src/urllib3/util/ssl_.py

```diff
@@ -251,19 +281,57 @@ def create_urllib3_context(
     context.verify_mode = cert_reqs
     # We ask for verification here but it may be disabled in HTTPSConnection.connect
     context.check_hostname = cert_reqs == ssl.CERT_REQUIRED
-    if hasattr(context, "hostname_checks_common_name"):
+    if sys.version_info >= (3, 7):
```

Comment:
This situation is more complex than this unfortunately (see https://github.com/urllib3/urllib3/issues/2192#issuecomment-821832963) so we can leave this as-is for now.

Reply:
Right, I see. Changed back to `hasattr` check.